src/Microsoft.AspNetCore.WebUtilities/WebEncoders.cs

```diff
@@ -48,50 +51,125 @@ public static byte[] Base64UrlDecode(string input, int offset, int count)
                 throw new ArgumentNullException(nameof(input));
             }
 
-            ValidateParameters(input.Length, offset, count);
+            ValidateParameters(input.Length, nameof(input), offset, count);
 
             // Special-case empty input
             if (count == 0)
             {
-                return new byte[0];
+                return EmptyBytes;
+            }
+
+            // Copy to an array large enough to Base64 characters (not just shorter Base64-URL-encoded form).
+            char[] completeBase64Array = new char[GetArraySizeRequiredToDecode(count)];
+            input.CopyTo(offset, completeBase64Array, 0, count);
+
+            return Base64UrlDecode(completeBase64Array, 0, count);
+        }
+
+        /// <summary>
+        /// Decodes a base64url-encoded <paramref name="input"/> into a <c>byte[]</c>.
+        /// </summary>
+        /// <param name="input">
+        /// The <see cref="char"/>s to decode. Array must be large enough to hold <paramref name="offset"/> and
+        /// <paramref name="count"/> characters as well as Base64 padding characters. Content is not preserved.
```

Comment:
Completely agree but it's not useful because the "output array" is always allocated within `Convert.FromBase64CharArray()`. One option would be
``` c#
public static byte[] Base64UrlDecode(
    string input, // All callers start w/ a string; char[] choice below was due only to reuse as an intermediate buffer.
    int offset,
    int count,
    char[] buffer) // Intermediate buffer of at least size GetArraySizeRequiredToDecode() returns.
{
    ...
}
```

Thoughts?